website/src/App.tsx

```diff
@@ -33,49 +33,24 @@ export const App = observer(class App extends React.Component {
       },
     });
 
-    if(AppState.loadingError){
-      return (
-        <Router>
-            <ThemeProvider theme={darkLightTheme}>
-              <CssBaseline />
-              <Navigation />
-              <Box component="div" m={2}>
-              <Error message={ AppState.loadingError } />
-              </Box>
-            </ThemeProvider>
-          </Router>
-      )
-    }
-
-    if (!AppState.info) {
-      return (
-          <Router>
-            <ThemeProvider theme={darkLightTheme}>
-              <CssBaseline />
-              <Navigation />
-              <Box component="div" m={2}>
-                <Loading />
-              </Box>
-            </ThemeProvider>
-          </Router>
-      );
-    }
-
-
     return (
       <Router>
         <ThemeProvider theme={darkLightTheme}>
           <CssBaseline />
           <Navigation />
           <Box component="div" m={2}>
-            <Routes>
-              <Route path="/" element={<YourDevices />} />
-              {AppState.info.isAdmin && (
-                <>
-                  <Route path="/admin/all-devices" element={<AllDevices />} />
-                </>
-              )}
-            </Routes>
+            {AppState.loadingError ? (
+              <Error message={AppState.loadingError} />
+            ) : (
+              !AppState.info ? (
+                <Loading />
+              ) : (
+                <Routes>
+                  <Route path="/" element={<YourDevices />} />
+                  {AppState.info.isAdmin && <Route path="/admin/all-devices" element={<AllDevices />} />}
+                </Routes>
+              )
+            )}
```

Comment:
Cool that we can reduce redundancy; but this nested branching is a bit hard to parse. What if we extract this part into a method like this?
```
...
         <Box component="div" m={2}>
            { this.pageContent() }
          </Box>
...
```
With a function like this:
```
  pageContent() {
    if (AppState.loadingError) {
      return (
          <Error message={AppState.loadingError} />
      );
    }
    else if (AppState.info) {
    ...

  }

```

Reply:
sounds good to me
[`19e2b98` (#605)](https://github.com/freifunkMUC/wg-access-server/pull/605/commits/19e2b98a8dabd1b2a5283cf68670f9f10b3f5cc3)